platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,193 @@
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.io.IOUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class SPMetadataParserTest {
+
+  private static Set<Binding> bindingSet;
+
+  private static final Map EMPTY_MAP = Collections.emptyMap();
+
+  private static String entityID;
+  private static EntityInformation testInfo;
+
+  private static String entityID2;
+  private static EntityInformation testInfo2;
+
+  private static final String METADATA_FILE = "SPMetadata";
+  private static final String METADATA_FILE_2 = "SPMetadata2";
+
+  private static final String CERTIFICATE_NAME = "Certificate.cert";
+
+  private static final String ENTITY_ID =
+      "http://localhost:80/spring-security-saml2-sample/saml/metadata";
+  private static final String ENTITY_ID_2 = "https://localhost:8993/services/saml/test";
+
+  private static final String REDIRECT_LOGOUT_URL =
+      "https://localhost:8993/services/saml/logout/redirect";
+  private static final String POST_LOGOUT_URL = "https://localhost:8993/services/saml/logout/post";
+  private static final String REDIRECT_ACS_URL =
+      "https://localhost:8993/services/saml/sso/redirect";
+  private static final String POST_ACS_URL = "https://localhost:8993/services/saml/sso/post";
+  private static final String SOAP_ACS_URL = "https://localhost:8993/services/saml/sso/soap";
+  private static String certificate;
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    // read Certificate file into certificate
+    certificate =
+        IOUtils.toString(
+            SPMetadataParserTest.class.getClassLoader().getResourceAsStream(CERTIFICATE_NAME));
+
+    // read SPMetadata file into spMetadata
+    List<String> spMetadata = new ArrayList<>();
+    spMetadata.add(
+        IOUtils.toString(
+            SPMetadataParserTest.class.getClassLoader().getResourceAsStream(METADATA_FILE)));
+    spMetadata.add(
+        IOUtils.toString(
+            SPMetadataParserTest.class.getClassLoader().getResourceAsStream(METADATA_FILE_2)));
+
+    // set up binding set
+    bindingSet = ImmutableSet.of(Binding.HTTP_POST, Binding.HTTP_REDIRECT, Binding.SOAP);
+
+    // set up metadata
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(spMetadata, bindingSet);
+    Iterator<Map.Entry<String, EntityInformation>> it = testMap.entrySet().iterator();
+
+    // get the first metadata
+    Map.Entry<String, EntityInformation> testEntry = it.next();
+    entityID = testEntry.getKey();
+    testInfo = testEntry.getValue();
+
+    // get the second metadata
+    testEntry = it.next();
+    entityID2 = testEntry.getKey();
+    testInfo2 = testEntry.getValue();
+  }
+
+  @Test
+  public void parseNullMetadata() throws Exception {
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(null, bindingSet);
+
+    assertThat(testMap, is(equalTo(EMPTY_MAP)));
+  }
+
+  @Test
+  public void testentityID2s() throws Exception {
```

Comment:
✏️ This test name is confusing. If you retrieve the `EntityInformation` objects by key instead of iterator (per my comment above), this test could just check that the two objects are not null and be named something like `testEntityInformationIsPresent`.